Source/WebGPU/WebGPU/UsdModelRenderer.swift

```diff
@@ -0,0 +1,439 @@
+                    index: UInt(index),
+                    semantic: DRVertexSemantic(rawValue: attribute.semantic) ?? .invalid,
+                    format: MTLVertexFormat(rawValue: UInt(attribute.format)) ?? .invalid,
+                    layoutIndex: DRUInteger(attribute.layoutIndex),
+                    offset: DRUInteger(attribute.offset)
+                )
+            }
+        }
+        if let vertexLayouts {
+            descriptor.vertexLayoutCount = vertexLayouts.count
+            for (index, layout) in vertexLayouts.enumerated() {
+                descriptor.setVertexLayout(
+                    layoutIndex: UInt(index),
+                    bufferIndex: DRUInteger(layout.bufferIndex),
+                    bufferOffset: DRUInteger(layout.bufferOffset),
+                    bufferStride: DRUInteger(layout.bufferStride)
+                )
+            }
+        }
+        return descriptor
+    }
+    #endif
+}
+
+@objc
+@implementation
+extension DDBridgeMeshPart {
+    var indexOffset: UInt
+    var indexCount: UInt
+    var topology: UInt
+    var materialIndex: UInt
+    var boundsMin: simd_float3
+    var boundsMax: simd_float3
+
+    init(
+        indexOffset: UInt,
+        indexCount: UInt,
+        topology: UInt,
+        materialIndex: UInt,
+        boundsMin: simd_float3,
+        boundsMax: simd_float3
+    ) {
+        self.indexOffset = indexOffset
+        self.indexCount = indexCount
+        self.topology = topology
+        self.materialIndex = materialIndex
+        self.boundsMin = boundsMin
+        self.boundsMax = boundsMax
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeSetPart {
+    var partIndex: Int
+    var part: DDBridgeMeshPart
+
+    init(
+        index: Int,
+        part: DDBridgeMeshPart
+    ) {
+        self.partIndex = index
+        self.part = part
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeSetRenderFlags {
+    var partIndex: Int
+
+    var renderFlags: UInt64
+
+    init(
+        index: Int,
+        renderFlags: UInt64
+    ) {
+        self.partIndex = index
+        self.renderFlags = renderFlags
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeReplaceVertices {
+    var bufferIndex: Int
+    var buffer: Data
+
+    init(
+        bufferIndex: Int,
+        buffer: Data
+    ) {
+        self.bufferIndex = bufferIndex
+        self.buffer = buffer
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeChainedFloat4x4 {
+    var transform: simd_float4x4
+    var next: DDBridgeChainedFloat4x4?
+}
+
+@objc
+@implementation
+extension DDBridgeUpdateMesh {
+    var partCount: Int
+    var parts: [DDBridgeSetPart]?
+    var renderFlags: [DDBridgeSetRenderFlags]?
+    var vertices: [DDBridgeReplaceVertices]?
+    var indices: Data?
+    var transform: simd_float4x4
+    var instanceTransforms: DDBridgeChainedFloat4x4? // array of float4x4
+    var materialIds: [UUID]?
+
+    override init() {
+        self.partCount = 0
```

Comment:
this init shouldnt exist; the initial values should just be set when creating the property